#161877953 Delete an article

victor-nweze-deel opened the pull request:
#### What does this PR do?
Enable users delete articles they create

#### Description of Task to be completed?
- Create a controller to delete articles
- Create route with proper validations
- Write integration tests
- Document endpoints

#### How should this be manually tested?
- Git clone this repository
- Cd into the `haven-ah-backend` folder
- Start up postman and hit the route `localhost:5000/api/v1/articles/:slug` with a `DELETE` request
- To test the entire application, run `npm test`

#### What are the relevant pivotal tracker stories?
[#161877953](https://www.pivotaltracker.com/story/show/161877953)

#### Screenshots (if appropriate)
*Attempting to delete an article you didn't create*
<img width="1216" alt="screen shot 2018-11-13 at 6 44 19 pm" src="https://user-images.githubusercontent.com/41582565/48432285-3bc83800-e774-11e8-8220-5d70669c7703.png">
*Deleting an article with the authorized token*
<img width="1219" alt="screen shot 2018-11-13 at 6 44 33 pm" src="https://user-images.githubusercontent.com/41582565/48432323-58647000-e774-11e8-9242-882591df9f9e.png">
*Attempting to delete a non-existent article*
<img width="1216" alt="screen shot 2018-11-13 at 6 43 43 pm" src="https://user-images.githubusercontent.com/41582565/48432353-69ad7c80-e774-11e8-9b6e-7e05061aba24.png">

ucheg6 commented:
Great Job Victor. This looks great

TheoOkafor commented on server/controllers/article.js:
```diff
@@ -150,6 +150,22 @@ class Article {
+    const { userid } = article.dataValues;
+    if (userid === userId) {
+      await articleRepo.deleteArticle(article.dataValues);
+      return goodHttpResponse(response, 202, 'Article deleted');
```
Is 202 the right status code for this?

victor-nweze-deel replied:
For a delete, we could have a general 200 status or a specific 202 or 204. Since I am returning a message it cant be 204. [So 202 is the most specific](https://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete).

victor-nweze-deel replied:
Yes it is. See [here](https://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete)

seyi-adeleke replied:
Ideally, this should be a 204 but since we're returning a payload 202 works as well.

TheoOkafor reviewed (commented):
Nice work @ProfJigsaw

seyi-adeleke commented on server/controllers/article.js:
```diff
@@ -150,6 +150,22 @@ class Article {
+      await articleRepo.deleteArticle(article.dataValues);
+      return goodHttpResponse(response, 202, 'Article deleted');
+    }
+    return goodHttpResponse(response, 401, 'You cannot deleted this article');
```
Can an admin delete an article?

victor-nweze-deel replied:
That was an oversight. I have fixed it. An admin can delete an article now.

victor-nweze-deel committed:
ft(articles): Delete an article

- Create a controller to delete articles
- Create route with proper validations
- Write integration tests
- Document endpoints
